packages/rancher-monitoring/templates/crd-template/templates/jobs.yaml

```diff
@@ -30,7 +30,12 @@ spec:
             {{- range $path, $_ :=  (.Files.Glob  "crd-manifest/**.yaml") }}
             {{- $crd := get (get ($.Files.Get $path | fromYaml) "metadata") "name" }}
             if [[ -n "$(kubectl get crd {{ $crd }} -o jsonpath='{.spec.preserveUnknownFields}')" ]]; then
-              kubectl patch crd {{ $crd }} -p '{"spec": {"preserveUnknownFields": false, "versions": [{"name": "v1", "served": false, "storage": true}]}}' --type="merge" || true;
+                if [[ -n "$(kubectl get crd {{ $crd }} -o jsonpath='{.spec.versions[0].schema}' 2>&1)" ]]; then
+                  patch="{\"spec\": {\"preserveUnknownFields\": false}}"
+                else
+                  patch="{\"spec\": {\"preserveUnknownFields\": false, \"versions\": [{\"name\": \"v1\", \"served\": false, \"storage\": true}]}}"
```

Comment:
We only need this patch when the schema does not exist since `preserveUnknownFields=false` requires a schema. This is the case for when Monitoring V2 installs onto a clean Rancher cluster, since Rancher does not define a schema.

However, when a schema exists, all we need to do is update `preserveUnknownFields`. This is the case for clusters that have enabled Monitoring V1 and then disabled it before installing Monitoring V2.

Reply:
Rather an if/else, should we set patch and then update it if the condition is met?

Reply:
Sure